libs/api/domains/license-service/src/lib/client/adr-license-client/adrLicenseService.api.ts

```diff
@@ -22,20 +24,34 @@ export class GenericAdrLicenseApi implements GenericLicenseClient<AdrDto> {
     private adrApi: AdrApi,
   ) {}
 
+  handleError(error: any): any {
+    this.logger.error('ADR license fetch failed', {
+      exception: error,
+      message: (error as Error)?.message,
+      category: LOG_CATEGORY,
+    })
+
+    throw new ApolloError(
+      'Failed to resolve request',
+      error?.message ?? error?.response?.message,
+    )
+  }
+
+  private handle4xx(error: FetchError) {
+    if (error.status === 403 || error.status === 404) {
+      return null
+    }
+    this.handleError(error)
+  }
+
   async fetchLicense(user: User) {
     let license: unknown
-
     try {
       license = await this.adrApi
         .withMiddleware(new AuthMiddleware(user as Auth))
         .getAdr()
     } catch (e) {
-      this.logger.error('ADR license fetch failed', {
-        exception: e,
-        message: (e as Error)?.message,
-        category: LOG_CATEGORY,
-      })
-      return null
+      this.handle4xx(e)
```

Comment:
This should be called `handleError` since it'll do both: ignore 403/404 error and log the non-4xx error